### ISSUE TITLE
Support overriding STDOUT/STDERR for WASI

### ISSUE THREAD

**Issue by Veykril** [wasm3/wasm3#147]
Allow the host to set the STDOUT/STDERR location that WASI uses. A simple use case for this is so that the host can capture the output of a wasm module.

**Comment by cjihrig** [comment]
I can't speak for the other WASI integrations in Wasm3, but uvwasi does support this. [This code](https://github.com/wasm3/wasm3/blob/5aba4e43bd17741e1c8dd0d6737cf071e5ee6913/source/m3_api_uvwasi.c#L474-L476) would need to set stdin, stdout, and stderr to the desired values. For reference, this [feature](https://github.com/nodejs/node/pull/33544) just landed yesterday in Node.js.

**Comment by vshymanskyy** [comment]
There's also another way to do it. It's a matter of intercepting the `wasi_*:fd_write`, `wasi_*:fd_read` calls. For example in `m3_api_uvwasi.c`, at `m3ApiRawFunction(m3_wasi_unstable_fd_read)`, we can have something like this:
> ![image](https://user-images.githubusercontent.com/1506708/83362715-0ad24700-a39c-11ea-92ca-443b7062616f.png)

Then `m3_vfs_read` checks if `fd == stdin`, and overrides the input. Same for `m3_vfs_write`.
This is what I did to implement the `Wasmer framebuffer` (and `WasmBoy` emulation):
https://twitter.com/wasm3_engine/status/1255535687375499265
But I had to handle `m3_wasi_unstable_path_open` as well (for stdin/stdout/stderr, this should not be needed).

Now the bigger question is, how this redirection should be implemented API-wise?
Would the Wasm3 library user need to register read/write callbacks?
Should they handle the `iovec` logic on their side, or should it be abstracted away?
